### DISCUSSION TITLE
How to handle user sessions with Prisma example?

### DISCUSSION THREAD

**Discussion by junaid33** [prisma/prisma#4575] [Q&A] [answered]
Hello,

I was looking thru Prisma examples and starting to mess with this one:

https://github.com/prisma/prisma-examples/tree/latest/typescript/rest-nextjs-api-routes-auth

It uses Next-auth for authentication. Next-auth returns a session where we can get the accessToken and user email. The example above uses this code:

```ts
  const session = await getSession({ req });
  if (!session) {
    res.statusCode = 403;
    return { props: { drafts: [] } };
  }

  const drafts = await prisma.post.findMany({
    where: {
      author: { email: session.user.email },
      published: false,
    },
    include: {
      author: {
        select: { name: true },
      },
    },
  });
```

Is it secure just to check if the session exists and use the email? Or would this be better:

```ts
  const session = await getSession({ req });
  if (!session) {
    res.statusCode = 403;
    return { props: { drafts: [] } };
  }

   const [ses] = await prisma.session.findMany({
      where: { accessToken: session?.user?.accessToken },
    });

  const drafts = await prisma.post.findMany({
    where: {
      author: { id: ses.userId },
      published: false,
    },
    include: {
      author: {
        select: { name: true },
      },
    },
  });
```
This example gets the session from Prisma and then gets the userId from that session. Which would be the most secure way?

**Comment by ryands17** [comment, marked as answer]
Hey @junaid33 👋 
As far as I know, Next Auth validates the session so that the user doesn't need to handle that from their end. Which is why I think the first way should be fine.

Also in the second way, you are running an extra query which might also impact the response time.

**Comment by junaid33** [comment]
Makes sense. Just wanted to make sure the email in the session can't be spoofed.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
